src/session/session_helper.c

```diff
@@ -134,3 +134,35 @@ __wt_session_dump(WT_SESSION_IMPL *session, WT_SESSION_IMPL *dump_session, bool
     __wt_scr_free(session, &buf);
     return (ret);
 }
+
+/*
+ * __wt_session_set_last_error --
+ *     Stores information about the last error to occur during this session.
+ */
+int
+__wt_session_set_last_error(
+  WT_SESSION *wt_session, int err, int sub_level_err, const char *err_msg_content)
+{
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session = (WT_SESSION_IMPL *)wt_session;
+
+    char *err_msg = session->err_info.err_msg;
+    size_t err_msg_size;
+
+    /* Ensure arguments are valid, and that session is not a mock */
+    WT_ASSERT(session, __wt_is_valid_sub_level_error(sub_level_err));
+    if (err_msg_content == NULL || wt_session->get_last_error == NULL)
+        return (0);
+
+    /* Free the last error message string, if it was allocated. */
+    __wt_free(session, err_msg);
+
+    /* Load error codes and message content into err_info. */
+    err_msg_size = (strlen(err_msg_content) + 1) * sizeof(char);
+    WT_ERR(__wt_malloc(session, err_msg_size, &err_msg));
```

Comment:
calloc would be more appriopate, to ensure buffer is clean and zero'ed out

Reply:
Done in [Pass WT_SESSION_IMPL into __wt_session_set_last_error instead of WT_S…](https://github.com/wiredtiger/wiredtiger/pull/11361/commits/6fe3e3c5ccc5ba957a2fb8dab7b7992243af183b)

Reply:
I think `malloc` is enough as we write it immediately after the memory is allocated, is that just a code consistency thing or there're edge cases here?